src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -81,7 +83,14 @@ public Command parseCommand(String userInput) throws ParseException {
             return new ListCommand();
 
         case MarkDeliveryCommand.COMMAND_WORD:
-            return new MarkDeliveryCommandParser().parse(arguments.trim());
+            if (arguments.trim().startsWith("-d")) {
+                return new MarkDeliveryCommandParser().parse(arguments.trim());
+            } else if (arguments.trim().startsWith("-s")) {
+                return new MarkSupplierCommandParser().parse(arguments.trim());
+            } else {
```

Comment:
i think this is good, and we should follow this for the delete commands as well where if the command does not start with -s or -d, we will return an invalid command feedback to the user